### DISCUSSION TITLE
Discussions

### DISCUSSION THREAD

**Discussion by Noa3** [Noa3/Cowtastic-Open-Cafe#2] [General]
you can chat here with others

**Comment by midiconch** [comment]
Awesome! Just found out you open sources this!

I'm a speedrunner with lots of Unity experience so I definitely have ideas I want to implement.

On the off chance anyone else is looking at this, to avoid duplicating their work, here's some of the stuff I wanna do when I next get some free time:

1. Fix the RNG seeding (last time I tried to set the RNG, it updated the seed value in the UI but didn't seem to actually seed the RNG. There was no way for me to get  even just the first drink order consistent.)
2. Modify everything that uses RNG in the following manner: when you seed the RNG, it then generates three values: a "drink order seed", an "event seed", and a "cosmetic seed". This means that if you always play on a set seed, the drink orders will always come in a set order (assuming you have the same unlocks at that point) that you can in theory memorize. And the random events will always happen at the same time relative to the start of the game.
3. Add some sort of option for key bindings. Speedrunning this game is great but it kills my arm. There needs to be a way to run it with no or little mouse movement.

Tbh just getting to work on a breast expansion project this polished is gonna be a blast.

**Reply by midiconch** [reply]
Oh

4. Maybe have some sort of feature where when you manually enter a seed, it remembers that you used that seed on that game mode so you don't have to enter it every time. Maybe a lock button next to the seed box?
5. A button in-game to wipe your data.

**Reply by Noa3** [reply]
Hi, i look forward to you changes.
You can fork the project and then ask for a merge request of your fork branch and Orginal Main Branch for the featurs to be implemented.
1. On the Statics.cs you can see the Logic for the most Randomgeneration, its a faster Custon Burst variation. feel free to experiment with it.
2. sounds interesting, maybe this would be the solution to the current system not working
3. the main part of this game, was to be playable with one hand. so people can....enjoy the game. if you have an interessting layout for what the input can work out, i will try it out and implement this from you
4. it should not hard to make this. you can save everything in the playerprefs.
5. is allready in the options, maybe i should rename this.

little extra, you need maybe to add your own UI backgrounds because the ones i used got buyed from the asset store and cant be uploaded in the public. the music is a smilar problem.

i work on the translation and refactroring of the UI Layout to fit longer texts. some logic got complete rewritten but in the moment the new files arent usable and doesnt make much sense so i didnt copy them yet.

i which you good luck and have fun.

**Reply by midiconch** [reply]
Ah, I guess I just missed the reset data option. I'll look for it again later.

Yeah, maybe I should look finding replacements for those assets.

6. This is a bit of a long shot, but maybe I could implement a replay function? Where it saves a file recording your input times and durations and you can click a button to watch an exact replay of it? Would be nice for speedrunning, or just for zero-hand enjoyment.

**Comment by Yellowscar** [comment]
Can't code for sh#t and never made a mod before but I'm a decent artist and wanna mod the barista, does anybody either have advice or wanna collab on a mod?
